code/modules/vending/vending.dm

```diff
@@ -633,11 +633,17 @@
 					var/atom/movable/vended = new product_path(src.get_output_location()) // changed from obj, because it could be a mob, THANKS VALUCHIMP
 					vended.layer = src.layer + 0.1 //So things stop spawning under the fukin thing
 					if(isitem(vended))
-						usr.put_in_hand_or_eject(vended) // try to eject it into the users hand, if we can
+						if(!get_dist(usr, src) > 1)
```

Comment:
Does that work? Isn't that always true?
(`!` has precedence over `>`)

Reply:
Fixed all issues you raised